src/components/HashButton.tsx

```diff
@@ -0,0 +1,129 @@
+import React from "react";
+import {
+  Box,
+  Button,
+  Typography,
+  Popover,
+  IconButton,
+  useTheme,
+} from "@mui/material";
+import {grey} from "../themes/colors/aptosColorPalette";
+import ChevronRightRoundedIcon from "@mui/icons-material/ChevronRightRounded";
+import ChevronLeftRoundedIcon from "@mui/icons-material/ChevronLeftRounded";
+import {truncateAddress} from "../pages/utils";
+
+interface HashButtonProps {
+  hash: string;
+}
+
+export default function HashButton({hash}: HashButtonProps) {
+  const [anchorEl, setAnchorEl] = React.useState<HTMLButtonElement | null>(
+    null,
+  );
+
+  const hashExpand = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.stopPropagation();
```

Comment:
Can we redo it with preventDefault() / defaultPrevented instead of stopPropagation()? https://css-tricks.com/dangers-stopping-event-propagation/

Reply:
I tried `preventDefault()`. However, it does not work. Seems like `preventDefault()` does not stop the event from bubbling up, so `rowClick` will be called in the row component. Maybe @moonclavedev has a better idea since he authored this part of code.

Reply:
Yeah it would require a check for event.defaultPrevented in the event handler for the row navigation. No big deal